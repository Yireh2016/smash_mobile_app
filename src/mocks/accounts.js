export default [
  {
    id: '5fd97b81f415a00008611b8e',
    institution_name: 'American Express',
    name: 'Credit card 3',
    current_balance: 1100,
    type: 'credit',
    subtype: 'credit card',
    updated_at: '2020-12-16T12:48:57.288Z',
    number: '1426',
    purchase_apr_percentage: null,
  },
  {
    id: '5fd97b81f415a00008611b8e',
    institution_name: 'American Express',
    name: 'Credit card 2',
    current_balance: 1900,
    type: 'credit',
    subtype: 'credit card',
    updated_at: '2020-12-16T12:48:57.291Z',
    number: '1501',
    purchase_apr_percentage: null,
  },
  {
    id: '5fd97b81f415a00008611b8e',
    institution_name: 'American Express',
    name: 'Credit card',
    current_balance: 3100,
    type: 'credit',
    subtype: 'credit card',
    updated_at: '2020-12-16T12:48:57.292Z',
    number: '2129',
    purchase_apr_percentage: null,
  },
  {
    id: '5fd97b81f415a00008611fa5',
    institution_name: 'American Express',
    name: 'Test',
    current_balance: 3100,
    type: 'depository',
    subtype: 'checking',
    updated_at: '2020-12-16T12:48:57.292Z',
    number: '2129',
    purchase_apr_percentage: null,
  },
];
