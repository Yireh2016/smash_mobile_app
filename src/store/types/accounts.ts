import { SET_ACCOUNTS, RESET_ACCOUNTS } from '../constants';

export enum AccountTypes {
  DEPOSITORY = 'depository',
}

export enum AccountSubTypes {
  CHECKINGS = 'checking',
  SAVINGS = 'savings',
}
export interface Account {
  institution_name: string;
  name: string;
  current_balance: number;
  subtype: string;
  type: string;
  updated_at: string;
  number: string;
  purchase_apr_percentage: number | null;
  id: string;
}

export interface SetAccountAction {
  type: typeof SET_ACCOUNTS | typeof RESET_ACCOUNTS;
  payload: Account[];
}
