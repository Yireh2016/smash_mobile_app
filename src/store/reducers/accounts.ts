import { SET_ACCOUNTS, RESET_ACCOUNTS } from '../constants';
import { SetAccountAction, Account } from '../types/accounts';

const initialState: Account[] | null = null;
const setAccount = (action: SetAccountAction) => action.payload;
const resetAccount = () => null;

export default (state = initialState, action: SetAccountAction) => {
  switch (action.type) {
    case SET_ACCOUNTS: {
      return setAccount(action);
    }
    case RESET_ACCOUNTS: {
      return resetAccount();
    }
    default:
      return state;
  }
};
