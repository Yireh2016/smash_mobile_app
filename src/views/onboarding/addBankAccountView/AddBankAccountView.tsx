/* eslint-disable react-native/no-inline-styles */
import * as React from 'react';
import { TouchableOpacity, View } from 'react-native';
import InfoLayout from '../../../layouts/infoLayout/InfoLayout';

import { MainParagraph, P1 } from '../../../components/texts/Texts';

import StepsProgressBar from '../../../components/stepsProgressBar/StepsProgressBar';

import ActionButton from '../../../components/actionButton/ActionButton';
import TotalDebt from '../../../components/totalDebt/TotalDebt';
import AddAccountsTable from '../../../components/addAccountsTable/AddAccountsTable';
import formatMoneyValue from '../../../helpers/formatMoneyValue/formatMoneyValue';
import theme from '../../../theme/theme';
import { Account } from '../../../store/types/accounts';
import PlaidBanksLinksViewWrapper, {
  PlaidLinkOnSuccessCallbackType,
} from '../../../views/plaidBanksLinksView/PlaidBanksLinksViewWrapper';

interface AddBankAccountProps {
  onNextPress: () => void;
  onAddAccountSuccess: PlaidLinkOnSuccessCallbackType;
  onSecurityPress: () => void;
  accounts: Account[] | null;
  totalDebt: number;
  goToSafeAccountConnect: () => void;
}

const AddBankAccount: React.FunctionComponent<AddBankAccountProps> = ({
  onNextPress,
  onAddAccountSuccess,
  onSecurityPress,
  goToSafeAccountConnect,
  accounts = [],
  totalDebt,
}) => {
  const footer = (
    <View
      style={{
        marginTop: 20,
        flexDirection: 'row',
        justifyContent: 'center',
      }}
    >
      <P1>Learn more about our </P1>
      <TouchableOpacity onPress={onSecurityPress}>
        <P1 isBold={true} color={theme.palette.secondary}>
          Security
        </P1>
      </TouchableOpacity>
    </View>
  );

  return (
    <InfoLayout onPress={onNextPress} actionLabel={'Next'} footer={footer}>
      <MainParagraph style={{ marginBottom: 16 }}>Steps 2 of 3</MainParagraph>
      <View style={{ marginBottom: 16 }}>
        <StepsProgressBar progress={200 / 3} />
      </View>
      <MainParagraph>
        Add all your bank accounts to recommend a recurring payment toward your
        cards.
      </MainParagraph>
      {totalDebt > 0 && (
        <View style={{ marginBottom: 20 }}>
          <TotalDebt
            title="Total bank accounts"
            totalDebt={formatMoneyValue(totalDebt)}
          />
        </View>
      )}
      {accounts && accounts.length > 0 && (
        <View>
          <AddAccountsTable accounts={accounts} />
        </View>
      )}
      <View style={{ marginTop: accounts && accounts.length > 0 ? 16 : 20 }}>
        {accounts && accounts.length > 0 ? (
          <PlaidBanksLinksViewWrapper onSuccessCallback={onAddAccountSuccess}>
            <ActionButton label="Add bank account" />
          </PlaidBanksLinksViewWrapper>
        ) : (
          <ActionButton
            label="Add bank account"
            onPress={goToSafeAccountConnect}
          />
        )}
      </View>
    </InfoLayout>
  );
};

export default AddBankAccount;
