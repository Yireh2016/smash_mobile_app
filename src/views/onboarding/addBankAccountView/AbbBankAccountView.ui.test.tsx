import * as React from 'react';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import renderer from 'react-test-renderer';
import accounts from '../../../mocks/accounts';
import AddBankAccountView from './AddBankAccountView';
const AddBankAccountViewProps = {
  onNextPress: () => {},
  onAddAccountSuccess: () => {},
  onSecurityPress: () => {},
  accounts,
  totalDebt: 1234,
};

describe('', () => {
  it('AddBankAccountView onboarding view with accounts renders correctly', () => {
    const Element = () => {
      return (
        <SafeAreaProvider>
          <AddBankAccountView {...AddBankAccountViewProps} />
        </SafeAreaProvider>
      );
    };

    const rendered = renderer.create(<Element />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('AddBankAccountView onboarding view without accounts renders correctly', () => {
    const AddBankAccountViewProps2 = JSON.parse(
      JSON.stringify(AddBankAccountViewProps)
    );
    AddBankAccountViewProps2.accounts = null;
    AddBankAccountViewProps2.totalDebt = 0;
    const Element = () => {
      return (
        <SafeAreaProvider>
          <AddBankAccountView {...AddBankAccountViewProps2} />
        </SafeAreaProvider>
      );
    };

    const rendered = renderer.create(<Element />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
