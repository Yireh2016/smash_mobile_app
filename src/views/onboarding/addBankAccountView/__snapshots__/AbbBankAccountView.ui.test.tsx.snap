// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddBankAccountView onboarding view with accounts renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`AddBankAccountView onboarding view without accounts renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
